app/test/e2e/layer-delete.spec.js

```diff
@@ -0,0 +1,124 @@
+        createMockDataset('123');
+
+        const datasetLayers = await requester.delete(`${datasetPrefix}/123/layer`);
+        datasetLayers.status.should.equal(401);
+        ensureCorrectError(datasetLayers.body, 'Not authorized');
+    });
+
+    it('DELETE /dataset/:dataset/layer - Delete layers by dataset with being authenticated not as MICROSERVICE should fail', async () => {
+        const { USER, ADMIN, MANAGER } = ROLES;
+
+        const userRequests = await Promise.all([await deleteLayers(USER), await deleteLayers(ADMIN), await deleteLayers(MANAGER)]);
+
+        const testResponse = (datasetLayers) => {
+            datasetLayers.status.should.equal(401);
+            ensureCorrectError(datasetLayers.body, 'Not authorized');
+        };
+
+        userRequests.map(testResponse);
+    });
+
+    it('DELETE /dataset/:dataset/layer - should delete layers in specific dataset', async () => {
+        const datasetLayers = await deleteLayers(ROLES.MICROSERVICE);
+        datasetLayers.status.should.equal(200);
+
+        const layers = await Layer.find({});
+        expect(layers).to.be.length(0);
+    });
+
+    it('DELETE /dataset/:dataset/layer/:layer - should return error not found, when dataset is not exist', async () => {
+        const datasetLayer = await requester.delete(`${datasetPrefix}/321/layer/123`);
+        datasetLayer.status.should.equal(404);
+        ensureCorrectError(datasetLayer.body, 'Dataset not found');
+    });
+
+    it('DELETE /dataset/:dataset/layer/:layer - Delete specific layer without being authenticated should fail', async () => {
+        createMockDataset('123');
+
+        const datasetLayer = await requester.delete(`${datasetPrefix}/123/layer`);
+        datasetLayer.status.should.equal(401);
+        ensureCorrectError(datasetLayer.body, 'Not authorized');
+    });
+
+    it('DELETE /dataset/:dataset/layer/:layer - Delete specific layer with being authenticated as USER should fail', async () => {
+        const datasetLayer = await deleteLayer(ROLES.USER, '123');
+        datasetLayer.status.should.equal(403);
+        ensureCorrectError(datasetLayer.body, 'Forbidden');
+    });
+
+    it('DELETE /dataset/:dataset/layer/:layer - should return error not found, when layer is not exist', async () => {
+        createMockDataset('123');
+        const layer = createLayer(['rw'], '123', '321');
+        await new Layer(layer).save();
+
+        const datasetLayer = await requester
+            .delete(`${datasetPrefix}/123/layer/123?loggedUser=${JSON.stringify(ROLES.ADMIN)}`)
+            .send();
+
+        datasetLayer.status.should.equal(404);
+        ensureCorrectError(datasetLayer.body, 'Layer with id \'123\' doesn\'t exist');
+    });
+
+    it('DELETE /dataset/:dataset/layer/:layer - should delete the specific layer in specific dataset', async () => {
+        const datasetLayer = await deleteLayer(ROLES.ADMIN, '123');
+        datasetLayer.status.should.equal(200);
```

Comment:
please validate the response body content, not just the code. I assume this delete endpoint returns the deleted object in the body, so please ensure it returns the correct object, with the correct structure.